README.md

```diff
@@ -299,6 +299,13 @@ for await (const { octokit, repository } of app.eachRepository.iterator({ instal
 await app.eachRepository({ installationId }, ({ octokit, repository }) => /* ... */)
 ```
 
+### `app.getInstallationUrl`
+
+```js
+const installationUrl = await app.getInstallationUrl(state);
```

Comment:
we should accept an object with keys. Besides state, we might want to accept a `target_id` argument. That way we can link directly to the page where a user is prompted to install a given app on a specific account. It's also the same URL to where an existing installation can be updated. Not the additional `/permissions` to make `target_id` work though

For example:

https://github.com/apps/gr2m/installations/new/permissions?target_id=35275088

is prompting me directly to install the gr2m app on @gr2m-sandbox

<img width="577" alt="image" src="https://github.com/octokit/app.js/assets/39992/0759e59f-02a5-4c91-88b0-745531fe2eb7">

I think there might be other query parameters like prompting a user to directly add an existing repository to an existing installation, someone would need to double check the code for that feature, and if we want to officially document these features or not

Reply:
Oh wow that's interesting - even more reason to have this method I guess! I tried looking for github docs on this but could only find [this](https://docs.github.com/en/apps/sharing-github-apps/sharing-your-github-app). Do you have a better link?

>  if we want to officially document these features or not

That's a good point - the fact that I can't find any docs on these options makes me wonder if they're stable. If you think `target_id` is here to stay I'm happy to add

> we should accept an object with keys

for sure - didn't want to create a whole new type that had to be exported unless I could find 2+ configuration options

Reply:
@gr2m updated - lmk if there are any other concerns